### PR TITLE
Replace poorly designed aggregation package with a better-integrated Aggregation builder

### PR DESCRIPTION
**Pull request by kdkeck**

### CODE REVIEW

**Comment by maxkremer** [comment]
At the risk of sounding defensive I don't see the strengths of this design over the current "poorly" designed implementation. It would introduce a breaking change for anyone using the current aggregation stuff. Further it replaces a well documented implementation with one that has almost no javadoc.

File: src/main/java/org/mongojack/Aggregation.java [hunk 1]
```diff
@@ -60,4 +71,700 @@ public DBObject getInitialOp() {
...
+        public static Expression<Number> add(Expression<Number>... numbers) {
+            return new OperatorExpression<Number>("$add", numbers);
+        }
+
+        public static Expression<Number> divide(Expression<Number> number1, Expression<Number> number2) {
+            return new OperatorExpression<Number>("$divide", number1, number2);
+        }
+
+        public static Expression<Number> mod(Expression<Number> number1, Expression<Number> number2) {
+            return new OperatorExpression<Number>("$divide", number1, number2);
```
Comment (kdkeck) [inline src/main/java/org/mongojack/Aggregation.java]: Yes, good catch, thanks.

**Comment by benmccann** [comment]
I'll go ahead and merge this now, but yes I think adding some javadocs wouldn't hurt nor do more tests generally hurt

**Comment by maxkremer** [comment]
I'm trying to integrate the new aggregation stuff in my code and I've run into a small stumbling block. I can't seem to group by empty or null. From the mongo docs:

__"The _id field is mandatory; however, you can specify an _id value of null to calculate accumulated values for all the input documents as a whole."__

The above is illustrated by the following two query variations:

`db.someCollection.aggregate({$group:{_id:{}, count:{$sum:1}}})` 
or
`db.someCollection.aggregate({$group:{_id:null, count:{$sum:1}}})`

@kdkeck can you explain how to achieve this using the API?

**Comment by kdkeck** [comment]
It might be worth adding a no-arguments `static group()` method to Aggregation for this case, but the following should both work:
```java
        Aggregation.group(Expression.object(new HashMap())).set("count", Group.count());
```
or
```java
        Aggregation.group(Expression.literal(null)).set("count", Group.count());
```

**Comment by maxkremer** [comment]
Yes that works. Thanks. I will try adding a no-arguments `static group()` as well as some badly needed javadocs

**Pull request merged**